text/0005-architecture-reviews.md

```diff
@@ -0,0 +1,135 @@
+# Architecture Review Process
+
+## Introduction
+
+Architectural Review decisions within the Firefox organization fall into two categories: "Roadmap" and “Design”.
+
+Product Reviews are not the scope of this process. This process is about architectural change, which should ultimately be in support of product features, but will generally have an effect on multiple product features. Architectural change can be growing new skills or in reducing large scale technical debt.
+
+The function of a **Roadmap Review** is to decide if a thing should be done. The goal is to bring together a packet of data to inform a management decision to provide resources to make the thing happen. A Roadmap Review should happen early in the process so that build time isn’t wasted on a "No" decision, but so that enough information is available to management.
+
+A **Design Review** is to decide if the thing should be done in a particular way. The goal is to ensure that the problem space is understood well enough that a realistic bug breakdown can be made, that those working on the problem all agree on the challenge, and that work can be prioritized to hit the hardest parts of the problem first.
```

Comment:
Hm. The reason I make that distinction is that without agreement on the outcome I think many DRs will end up arguing about the problem, not the solution — that is, if you don't think we should be going to the moon, we're going to spend time arguing about that, not about whether my space elevator is better or worse than a shuttle, no? And if there isn't already agreement on the outcome, all of my research into nanowire elevators is wasted.

So I've been reading this document as:

* RR: agree on the problem, at a high level, and a resourcing roadmap to address it.
* DR: given an agreed problem, narrow down a defined outcome and an engineering path to get there.